File: erlang/hamming/src/hamming.erl
-module(hamming).

-export([distance/2]).

distance(Strand1, Strand2) when length(Strand1) =:= length(Strand2) ->
    distance(Strand1, Strand2, 0);
distance(_, _) ->
    {error, "left and right strands must be of equal length"}.

%% @private
distance([], [], C) ->
    C;
distance([H | T1], [H | T2], C) ->
    distance(T1, T2, C);
distance([_ | T1], [_ | T2], C) ->
    distance(T1, T2, C + 1).

File: kotlin/hamming/src/main/kotlin/Hamming.kt
package Hamming

fun compute(s1: String, s2: String): Int {
    require (s1.length == s2.length) {
        "left and right strands must be of equal length."
    }
    return s1.zip(s2).filter({ (c1, c2) -> c1 != c2 }).count()
}
